### PR TITLE
Internal Affairs actually works as intended now

### LINKED ISSUE #53404: IAA aren't stealing each other's objectives?
https://github.com/tgstation/tgstation/blob/master/code/modules/antagonists/traitor/IAA/internal_affairs.dm

Apparently, IAA agents are supposed to steal their targets' objectives when they kill them. Once they become the last IAA agent left on the station, they're then supposed to receive a license to murderbone.

From what I've played of (and seen of) the role on Fulp, I've never seen either of these two things happen, and I can't figure out _why_. Yeah, yeah, it's an unsupported gamemode and all that, but hey, maybe if this bug gets fixed, IAA will become distinct enough from normal traitors to bring back as its own mode.

### PR DESCRIPTION
**Pull request by SgtHunk**
## About The Pull Request

Hey you. Yes you. Did you know that IAA (Internal Affairs Agent) has been BROKEN this entire time? That's right. This gamemode has been BROKEN. It DIDN'T WORK. You never got assigned a target past your initial one, even though the code intended to. Instead, you were left with your initial target AAAAND if somehow, by miracle, all the fellow IAAs died, you'd get DAGD.
Oh, and to top it off, the malf AI had the wrong survive objective, so they always redtexted no matter what.

tl;dr Ports the IAA fix from https://github.com/BeeStation/BeeStation-Hornet/pull/3623 and AI fix from https://github.com/BeeStation/BeeStation-Hornet/pull/2376
Closes https://github.com/tgstation/tgstation/issues/53404

## Was this tested?

Yes, it was. Tested on a live 90+ pop server (since you can't test IAA on local by yourself.)

image
image
image
image

## Changelog
:cl:
fix: The Internal Affairs gamemode now actually works as intended, getting the target of the agent they just assassinated. Get to work, agents.
fix: IAA AIs will no longer always redtext their survive objective.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
oh my god finally

**Comment by ATH1909** [comment]
thank you, 3d glasses cat man

File: code/modules/antagonists/traitor/IAA/internal_affairs.dm [hunk 1]
```diff
@@ -13,6 +13,9 @@
 	var/last_man_standing = FALSE
 	var/list/datum/mind/targets_stolen
 
+/datum/antagonist/traitor/internal_affairs/New()
+	. = ..()
+	targets_stolen = list()
```
Comment (Ghommie) [inline code/modules/antagonists/traitor/IAA/internal_affairs.dm:18]:
```suggestion
	var/list/datum/mind/targets_stolen = list()
```

**Comment by bloons3** [comment]
Does this fix the pinpointer or does it only point to the first target?

**Comment by SgtHunk** [comment]
Only points at first target AFAIK, Though if it changed target it would be a bit inconsistent in case an earlier target gets revived

**Comment by bloons3** [comment]
IAA "magically" knows when your previous target that was dead is alive again, so it might be worthwhile to have it point to the first alive person in your objective list.

Otherwise it becomes very tedious looking for a laundry list of people across the entire station.

**Comment by SgtHunk** [comment]
It might be a bit out of scope for this PR, but it is something do-able
Could add it as a qol by adding an action button that allows you to switch your pinpointer's target if I have the time for it, but this PR is done as is

**Comment by ATH1909** [comment]
> It might be a bit out of scope for this PR, but it is something do-able
> Could add it as a qol by adding an action button that allows you to switch your pinpointer's target if I have the time for it, but this PR is done as is

How would this be out of scope?

I don't think the action button would even be needed; just make the proc that notifies you when one of your targets has been revived also change your pinpointer's target to that target.

File: code/modules/antagonists/traitor/IAA/internal_affairs.dm [hunk 2]
```diff
@@ -11,8 +11,7 @@
 	antagpanel_category = "IAA"
 	var/syndicate = FALSE
 	var/last_man_standing = FALSE
-	var/list/datum/mind/targets_stolen
-
+	var/list/datum/mind/targets_stolen = list()
```
Comment (Mothblocks) [inline code/modules/antagonists/traitor/IAA/internal_affairs.dm:14]: What's the point of this, why not keep it as a lazy list?
